### PR TITLE
fix: diamond problem

### PR DESCRIPTION
**Pull request by Eddort**
This pull request resolves the diamond dependency problem in 2 steps:
1. Listen to only writable stores
2. Bind callback with arguments (**run** function)

### CODE REVIEW

File: create-derived/index.test.ts [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/restrict-plus-operands */
```
Comment (ai) [inline create-derived/index.test.ts:1]: You do not need to disable this ESLint rule. Just use:

```js
    return `a${count}`
```

**Comment by ai** [comment]
1. Can I ask you to remove Diamond Problem note from the docs? I will make a link to PR in Twitter, it will be nice to fix the docs here.
2. Can you update Size Limit config and max size at the beginning of the docs? The size increase is low, great work. Just to make PR green to promote in Twitter.

File: create-derived/index.js [hunk 2]
```diff
@@ -1,20 +1,43 @@
...
+  )
+]
 
 export function createDerived(stores, cb) {
   if (!Array.isArray(stores)) stores = [stores]
+  let deps = collectWritable(stores)
+
+  let run = () => {
+    let values = stores.map(store =>
+      store.wrapper ? store.wrapper() : getValue(store)
```
Comment (ai) [inline create-derived/index.js:19]: Who can set `store.wrapper`?

File: create-derived/index.test.ts (hunk 1 above)
Comment (Eddort) [reply create-derived/index.test.ts]: Sure, it was a prototype

File: create-derived/index.js (hunk 2 above)
Comment (Eddort) [reply create-derived/index.js]: Resolved. I succeed to reduce the additional size from 54 to 35 B

Comment (ai) [reply create-derived/index.js]: 😍

**Review by ai — APPROVED** [review]

**Comment by ai** [comment]
Looks good. Will merge it when I will finish unpacking (relocation is hard 😅).

**Pull request merged**